File: Beecrowd/8-Geometria_Computacional/1124-Elevador-8/1124.py
import sys
import math

for linha in sys.stdin:
    L, C, R1, R2 = map(int, linha.split())
    
    if L == 0 and C == 0 and R1 == 0 and R2 == 0:
        break


    if 2 * R1 > L or 2 * R1 > C or 2 * R2 > L or 2 * R2 > C:
        print('N')
        continue


    distancia = math.sqrt((L - R1 - R2) ** 2 + (C - R1 - R2) ** 2)


    if distancia >= (R1 + R2):
        print('S')
    else:
        print('N')

File: Beecrowd/8-Geometria_Computacional/1124-Elevador-8/1124.js
var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = input.trim().split('\n');

for (let line of lines){
    var [l, c, r1, r2] = line.split(' ').map(x => parseInt(x));
    if(l==0 && c ==0 && r1 ==0 && r2 ==0)break;
    if(2*r1>l || 2*r1>c || 2*r2>l || 2*r2>c){
        console.log("N");
        continue
    }
    if(Math.sqrt(Math.pow(l-r1-r2, 2)+Math.pow(c-r2-r1, 2)) >= r1+r2){
        console.log("S");
        continue;
    }
    console.log("N");
}
